### PR TITLE
CMakeLists for each Service

### LINKED ISSUE #1523: Per-service CMakeLists.txt
Example: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/identity has a CMakeLists.txt file

Needed for: 
* Core
* Template

### PR DESCRIPTION
**Pull request by RickWinter**
Fixes #1523 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/core/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.12)
```
Comment (ahsonkhan) [inline sdk/core/CMakeLists.txt:4]: Make it consistent with other cmakelists and docs.
https://github.com/Azure/azure-sdk-for-cpp/blob/f285aaead94be16bd3a7af8ac6565aeb4d55a525/CONTRIBUTING.md#cmake
```suggestion
cmake_minimum_required (VERSION 3.13)
```

Identity is the only one that is inconsistent, which we should update as well:
https://github.com/Azure/azure-sdk-for-cpp/blob/a38de03e7b42cdf1f6df61dae86127a9bc4554e7/sdk/identity/CMakeLists.txt#L4

File: sdk/template/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.12)
```
Comment (ahsonkhan) [inline sdk/template/CMakeLists.txt:4]:
```suggestion
cmake_minimum_required (VERSION 3.13)
```

File: sdk/template/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+project (azure-template LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+add_subdirectory(azure-template)
+# add_subdirectory(performance-stress)
```
Comment (ahsonkhan) [inline sdk/template/CMakeLists.txt:12]: This doesn't exist under `/template`
```suggestion
```

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Bump cmake to 3.13 (in here and in identity), and remove perf stress comment from template. Otherwise, looks good.

File: sdk/template/CMakeLists.txt (hunk 3 above)
Comment (RickWinter) [reply sdk/template/CMakeLists.txt:12]: That's why its commented out :).  Fixed

**Comment by RickWinter** [comment]
@ahsonkhan This needs to go in tonight.  Please clear your blocking request.

**Review by vhvb1989 — APPROVED** [review]

**Review by ahsonkhan — APPROVED** [review]

**Comment by ahsonkhan** [comment]
That was quick :)

**Pull request merged**